sympy/core/compatibility.py

```diff
@@ -6,131 +6,16 @@
 
 import operator
 from collections import defaultdict
-from time import perf_counter as clock
-
-from sympy.external.gmpy import SYMPY_INTS, HAS_GMPY, GROUND_TYPES, gmpy
-
 
 """
-Python 2 and Python 3 compatible imports
-
-String and Unicode compatible changes:
-    * `unicode()` removed in Python 3, import `unicode` for Python 2/3
-      compatible function
-    * Use `u()` for escaped unicode sequences (e.g. u'\u2020' -> u('\u2020'))
-    * Use `u_decode()` to decode utf-8 formatted unicode strings
-
-Renamed function attributes:
-    * Python 2 `.func_code`, Python 3 `.__func__`, access with
-      `get_function_code()`
-    * Python 2 `.func_globals`, Python 3 `.__globals__`, access with
-      `get_function_globals()`
-    * Python 2 `.func_name`, Python 3 `.__name__`, access with
-      `get_function_name()`
-
-Moved modules:
-    * `reduce()`
-    * `StringIO()`
-    * `cStringIO()` (same as `StingIO()` in Python 3)
-    * Python 2 `__builtin__`, access with Python 3 name, `builtins`
-
-exec:
-    * Use `exec_()`, with parameters `exec_(code, globs=None, locs=None)`
-
-Metaclasses:
-    * Use `with_metaclass()`, examples below
-        * Define class `Foo` with metaclass `Meta`, and no parent:
-            class Foo(with_metaclass(Meta)):
-                pass
-        * Define class `Foo` with metaclass `Meta` and parent class `Bar`:
-            class Foo(with_metaclass(Meta, Bar)):
-                pass
+Some utility functios origniating from Python 2 and Python 3 compatible imports.
 """
 
 __all__ = [
-    'PY3', 'int_info', 'SYMPY_INTS', 'clock',
-    'unicode', 'u_decode', 'get_function_code', 'gmpy',
-    'get_function_globals', 'get_function_name', 'builtins', 'reduce',
-    'StringIO', 'cStringIO', 'exec_', 'Mapping', 'Callable',
-    'MutableMapping', 'MutableSet', 'Iterable', 'Hashable', 'unwrap',
-    'accumulate', 'with_metaclass', 'NotIterable', 'iterable', 'is_sequence',
-    'as_int', 'default_sort_key', 'ordered', 'GROUND_TYPES', 'HAS_GMPY',
+    'NotIterable', 'iterable', 'is_sequence',
+    'as_int', 'default_sort_key', 'ordered',
```

Comment:
https://github.com/sympy/sympy/blob/414fc3df3090fe67afd6c8312e4ebf202124d308/sympy/utilities/iterables.py#L11-L13

Adding `NotIterable` to that list and moving them there?